OpenRA.Mods.Common/Widgets/TooltipContainerWidget.cs

```diff
@@ -25,23 +25,48 @@ public class TooltipContainerWidget : Widget
 
 		public Action BeforeRender = Nothing;
 		public int TooltipDelayMilliseconds = 200;
+		readonly Widget container;
 		Widget tooltip;
 		int nextToken = 1;
 		int currentToken;
+		string id;
+		WidgetArgs widgetArgs;
 
 		public TooltipContainerWidget()
 		{
+			container = new ContainerWidget();
+			AddChild(container);
+
 			graphicSettings = Game.Settings.Graphics;
-			IsVisible = () => Game.RunTime > Viewport.LastMoveRunTime + TooltipDelayMilliseconds;
+			IsVisible = () =>
+			{
+				// PERF: Only load widget once visible.
+				var visible = Game.RunTime > Viewport.LastMoveRunTime + TooltipDelayMilliseconds;
+				if (visible)
+					LoadWidget();
+
+				return visible;
+			};
+		}
+
+		bool LoadWidget()
```

Comment:
Why does this return a `bool`?

Reply:
initially i had the pos property/function load the widget - it currently checks to see if tooltip is not null. later assumed the pos property/function would only be called if visible.or that the return pos would not matter if not visible.

so could become void.